### PR TITLE
Camera reset with GPIO

### PR DESCRIPTION
**Pull request by CptTony57**

### CODE REVIEW

File: module/input/Camera/src/Camera.cpp [hunk 1]
```diff
@@ -66,13 +66,27 @@ namespace input {
         	}
         });
 
-        // When we shutdown, we must tell our camera class to close (stop streaming)
-        on<Shutdown>().then([this] {
+            on<Shutdown>().then([this] {
```
Comment (TrentHouliston) [inline module/input/Camera/src/Camera.cpp]: You should install the clang-format plugin so the indentation stops breaking everywhere

File: module/input/Camera/src/V4L2Camera.cpp [hunk 2]
```diff
@@ -1,55 +1,106 @@
...
+#include "extension/FileWatch.h"
 
 namespace module
 {
-    namespace input
-    {
-        using extension::Configuration;
+	namespace input
+	{
+		using extension::Configuration;
```
Comment (TrentHouliston) [inline module/input/Camera/src/V4L2Camera.cpp]: Broken indentation

File: module/input/Camera/src/V4L2Camera.cpp [hunk 3]
```diff
@@ -78,7 +129,7 @@ namespace module
 
                 log("Initialising driver for camera", deviceID);
 
-                V4L2Camera camera(config, deviceID, cameraCount);
+                //V4L2Camera camera(config, deviceID);
```
Comment (TrentHouliston) [inline module/input/Camera/src/V4L2Camera.cpp]: No need for commented out code

**Comment by TrentHouliston** [comment]
Dismiss my review and merge once @Bidski has reviewed it and approved it. He is in a better position to make sure this is all correct.

File: module/input/Camera/src/V4L2Camera.cpp [hunk 4]
```diff
@@ -1,132 +1,488 @@
...
-                                if (camera.second.setSetting(it->second, setting.second.as<int>()) == false)
-                                {
+                            if (it != settings.end()) {
+                                if (camera.second.setSetting(it->second, setting.second.as<int>()) == false) {
                                     log<NUClear::DEBUG>("Failed to set", it->first, "on camera", camera.first);
+                                    errorCount++;
                                 }
                             }
                         }
+                        if (errorCount > 40) {
```
Comment (Bidski) [inline module/input/Camera/src/V4L2Camera.cpp]: Make this a config parameter

File: module/input/Camera/src/V4L2Camera.cpp [hunk 5]
```diff
@@ -1,132 +1,488 @@
...
+        // Check if we managed to open our file descriptor
+        int resetCount = 0;
+
+        while (fd < 0 && resetCount < 10) {
+            std::cout << "Toggling GPIO" << std::endl;
+            system("/home/nubots/gpio_toggle.sh");
+            fd = open(deviceID.c_str(), O_RDWR);
+            resetCount++;
+
+            /*std::ofstream gpio;
```
Comment (Bidski) [inline module/input/Camera/src/V4L2Camera.cpp]: Delete this block of comments if they are no longer required.

**Pull request merged**